src/app.ts

```diff
@@ -13,14 +13,35 @@ interface TemperatureSummary {
   average: number
 }
 
+const data: TemperatureReading[] = []
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  readings.forEach((item) => {
+    data.push(item)
+  })
 }
 
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
-  //add here your code
+  const temperatures = data
+    .filter(
+      (item) =>
+        item.city === city && item.time.toDateString() === date.toDateString(),
+    )
+    .map((item) => item.temperature)
+
+  if (temperatures.length !== 0) {
```

Comment:
I usually prefer an early return, breaking the code as soon as the condition is not satisfied so in this case you could do

 if (temperatures.length === 0) {
    return null
  }
  
 return { // The rest of the object

Reply:
ok, I got it.